core/runtime/src/main/java/io/quarkus/runtime/shutdown/ShutdownRecorder.java

```diff
@@ -30,6 +30,9 @@ public void setListeners(List<ShutdownListener> listeners) {
     }
 
     public static void runShutdown() {
+        if (shutdownListeners == null) {
+            return;
+        }
```

Comment:
Best to move it under the debug log

Reply:
I actually think this one makes a little more sense